src/lib/pci_virtio_sock.c

```diff
@@ -130,6 +130,11 @@ static int pci_vtsock_debug = 0;
 
 /* XXX need to use rx and tx more consistently */
 
+struct vtsock_config_hdr {
+	uint32_t len;
+	uint32_t version;
+};
```

Comment:
You _could_ move the addition of `type` to the commit which introduces the protocol and `struct vtsock_config_hdr` and include a dummy:
```
	switch (hdr->type) {
	default:
		fprintf(stderr, "config: ERROR "
			"unknown configuration type %d\n", hdr->type);
	}
```
where your eventual switch would be, but it's probably not worth the rebase faff now that you've explained the temporary inconsistency in that one commit here